experiment/autobumper/bumper/bumper.go

```diff
@@ -161,36 +156,39 @@ func (gc GitCommand) getCommand() string {
 }
 
 func validateOptions(o *Options) error {
+	if len(o.Prefixes) == 0 {
+		return fmt.Errorf("Must have at least one Prefix specified")
+	}
 	if !o.SkipPullRequest && o.GitHubToken == "" {
-		return fmt.Errorf("--github-token is mandatory when --skip-pull-request is false")
+		return fmt.Errorf("github-token is mandatory when skip-pull-request is false")
 	}
-	if !o.SkipPullRequest && (o.GitHubOrg == "" || o.GitHubRepo == "") {
-		return fmt.Errorf("--github-org and --github-repo are mandatory when --skip-pull-request is false")
+	if (o.GitEmail == "") != (o.GitName == "") {
+		return fmt.Errorf("git-name and git-email must be specified together")
 	}
 	if !o.SkipPullRequest && o.RemoteBranch == "" {
-		return fmt.Errorf("--remote-branch cannot be empty when --skip-pull-request is false")
-	}
-	if (o.GitEmail == "") != (o.GitName == "") {
-		return fmt.Errorf("--git-name and --git-email must be specified together")
+		return fmt.Errorf("remote-branch cannot be empty when skip-pull-request is false")
 	}
 
+	if !o.SkipPullRequest && (o.GitHubOrg == "" || o.GitHubRepo == "") {
+		return fmt.Errorf("github-org and github-repo are mandatory when skip-pull-request is false")
+	}
+	if len(o.IncludedConfigPaths) == 0 {
+		return fmt.Errorf("include-config-paths is mandatory")
+	}
 	if o.TargetVersion != latestVersion && o.TargetVersion != upstreamVersion &&
 		o.TargetVersion != upstreamStagingVersion && !tagRegexp.MatchString(o.TargetVersion) {
-		logrus.Warnf("Warning: --target-version is not one of %v so it might not work properly.",
+		logrus.Warnf("Warning: target-version is not one of %v so it might not work properly.",
 			[]string{latestVersion, upstreamVersion, upstreamStagingVersion, tagVersion})
 	}
-	if !o.BumpProwImages && !o.BumpBoskosImages && !o.BumpTestImages {
-		return fmt.Errorf("at least one of --bump-prow-images, --bump-boskos-images and --bump-test-images must be specified")
-	}
-	if (o.BumpProwImages || o.BumpBoskosImages) && o.BumpTestImages && o.TargetVersion != latestVersion {
-		return fmt.Errorf("--target-version must be latest if you want to bump both prow/boskos and test images")
-	}
-	if o.BumpTestImages && (o.TargetVersion == upstreamVersion || o.TargetVersion == upstreamStagingVersion) {
-		return fmt.Errorf("%q and %q versions can only be specified to bump prow/boskos images", upstreamVersion, upstreamStagingVersion)
+	if o.TargetVersion == upstreamVersion {
+		for _, prefix := range o.Prefixes {
+			if prefix.RefConfigFile == "" || prefix.StagingRefConfigFile == "" {
+				return fmt.Errorf("target Version can't be 'upstream' without RefConfigFile and StagingRefConfigFile for each prefix")
+			}
+		}
 	}
-
-	if len(o.IncludedConfigPaths) == 0 {
-		return fmt.Errorf("--include-config-paths is mandatory")
+	if o.TargetVersion == upstreamVersion && o.UpstreamURLBase == "" {
```

Comment:
Can we default this UpstreamURLBase as current repo?

Reply:
Sorry for not making it clear, I meant set the default to `https://raw.githubusercontent.com/kubernetes/test-infra/master` since this is where the tool lives

Reply:
That makes sense. Done